### PR TITLE
Action view bugs

### LINKED ISSUE #249: Action View Issues
2 issues:

- The date of the action taken is not logged anywhere, only the date of the action item created
- Cannot view the full content of a note that extends the input element (possibly create a module to view whole note)

### PR DESCRIPTION
**Pull request by pdotsani**
Resolves #249 

@dting 

- Created view note module
- Experimented with create note module, resorted to vertical expanding textarea
- Created filter to parse strings in note view

Some styles have been applied inline, will add these to stylesheets after review.

### CODE REVIEW

File: client/components/modal/modal.service.js [hunk 1]
```diff
@@ -217,8 +217,9 @@ app.factory('Modal', function($rootScope, $uibModal) {
           title: title,
           templateUrl: templateUrl,
           submitFn: function(form, model) {
-            cb(model).$promise.then(function() {
+            cb(model).$promise.then(function(data) {
               confirmDelete.close();
+              console.log(data);
             }, function(err) {
```
Comment (pdotsani) [inline client/components/modal/modal.service.js]: Oops... this must have been something I left when I was experimenting with modals, will delete after review.

File: client/components/filters/filters.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict';
```
Comment (dting) [inline client/components/filters/filters.js]: I don't think we need this filter. This can be doing using css without a hardcoded width.

https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/

in our case:

student.scss 
```
.student {
  display: flex;
  flex-direction: column;
  height: 100%;
  @media(min-width: 768px) {
    .info {
      height: 108px;
      display: flex;
      flex-direction: column;
      justify-content: space-around;
    }
  }
  @media(max-width: 768px) {
    .info {
      padding: 15px 0;
      min-height: 324px;
    }
  }
  .interventions,
  .outreaches {
    flex: 1 0 0;
    padding: 15px 0;
    .toggle {
      cursor: pointer;
    }
    .invisible-menu {
      background: none !important;
      border: none !important;
      box-shadow: none !important;
    }
    .archived {
      background-color: lightgrey;
    }
    .truncate {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
}
```

interventions.html
```
...
          <div class="input-group">
            <input type="text" class="form-control truncate" readonly value="{{note.note}}"/>
            <span class="input-group-addon">{{note.date | date:'yyyy-MM-dd'}}</span>
          </div>
...
```
outreaches.html
```
...
          <div class="input-group">
            <input type="text" class="form-control truncate" readonly value="{{note.note}}"/>
            <span class="input-group-addon">{{note.date | date:'yyyy-MM-dd'}}</span>
            <span class="input-group-btn">
              <button class="btn btn-info" ng-click="viewNote(note, student)">
                <i class="fa fa-search"></i>
              </button>
            </span>
          </div>
...
```

File: client/app/main/student/partial/outreaches.html [hunk 3]
```diff
@@ -39,8 +39,8 @@
         <dt>Notes</dt>
         <dd>
           <div class="input-group">
-            <input type="text" class="form-control" ng-model="outreach.newNote"/>
-            <span class="input-group-btn">
+            <textarea type="text" class="form-control" ng-model="outreach.newNote" cols="30" rows="2" style="resize: vertical;"></textarea>
```
Comment (dting) [inline client/app/main/student/partial/outreaches.html]: I think we should revert this change for now. We aren't supporting multi line notes:

screen shot 2016-03-16 at 4 25 42 pm

Using a textarea will give an impression that notes can multiple lines. Then after adding the notes the lines will be concatenated.

File: client/app/main/student/partial/outreaches.html [hunk 4]
```diff
@@ -39,8 +39,8 @@
...
+            <span class="input-group-btn" style="vertical-align: top;">
```
Comment (dting) [inline client/app/main/student/partial/outreaches.html]: Avoid using inline styles.

File: client/app/main/student/partial/outreaches.html [hunk 5]
```diff
@@ -51,8 +51,13 @@
         </dd>
         <dd ng-repeat="note in outreach.notes | orderBy:$index:true">
           <div class="input-group">
-            <input type="text" class="form-control" readonly value="{{note.note}}"/>
+            <input type="text" class="form-control" readonly value="{{note.note | parseString}}"/>
```
Comment (dting) [inline client/app/main/student/partial/outreaches.html]: `parseString` filter can be removed. See note below.

File: client/app/main/student/partial/outreaches.html [hunk 6]
```diff
@@ -51,8 +51,13 @@
...
             <span class="input-group-addon">{{note.date | date:'yyyy-MM-dd'}}</span>
+            <span class="input-group-btn">
+              <button class="btn btn-info" ng-click="viewNote(note, student)">
+                <i class="fa fa-search"></i>
```
Comment (dting) [inline client/app/main/student/partial/outreaches.html]: lets use `<i class="fa fa-ellipsis-h"></i>` rather than search so it doesn't look like a search.

File: client/app/main/student/partial/outreaches.html (hunk 3 above)
Comment (dting) [inline client/app/main/student/partial/outreaches.html]: That being said, multi line support would be nice. We could look into that for a different change list.

File: client/components/filters/filters.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+var app = angular.module('app');
+
+app.filter('parseString', function() {
```
Comment (dting) [inline client/components/filters/filters.js]: `parseString` doesn't really describe the function of this filter. `truncate` would probably be a better name.

File: client/app/main/student/student.controller.js [hunk 8]
```diff
@@ -144,6 +144,17 @@ function StudentOutreachesCtrl($scope, Outreach, Modal, toastr) {
       deleteFn);
   };
 
+  $scope.viewNote = function(note, student) {
```
Comment (dting) [inline client/app/main/student/student.controller.js]: This probably needs to be added to the `StudentInterventionsCtrl` as well.

**Comment by dting** [comment]
I made some comments. Let me know if you have any questions or after you make the changes so I can review again. Thanks.

**Comment by pdotsani** [comment]
@dting 

Changes made. Will squash commits after review.

Thanks!

**Comment by dting** [comment]
Let's merge up to master and then review since some of the merge is in direct conflict. Let me know when to take another look.

File: client/app/main/student/partial/outreaches.html [hunk 9]
```diff
@@ -51,8 +49,13 @@
         </dd>
         <dd ng-repeat="note in outreach.notes | orderBy:$index:true">
           <div class="input-group">
-            <input type="text" class="form-control" readonly value="{{note.note}}"/>
-            <span class="input-group-addon">{{note.date | date:'MM/dd/yy'}}</span>
+            <input type="text" class="form-control truncate" readonly value="{{note.note}}"/>
+            <span class="input-group-addon">{{note.date | date:'yyyy-MM-dd'}}</span>
```
Comment (dting) [inline client/app/main/student/partial/outreaches.html]: Let's change this date back to 'MM/dd/yy'

This matches 'shortDate' format for uib.

File: client/app/main/student/partial/interventions.html [hunk 10]
```diff
@@ -44,8 +42,13 @@
         </dd>
         <dd ng-repeat="note in intervention.notes | orderBy:$index:true">
           <div class="input-group">
-            <input type="text" class="form-control" readonly value="{{note.note}}"/>
-            <span class="input-group-addon">{{note.date | date:'MM/dd/yy'}}</span>
+            <input type="text" class="form-control truncate" readonly value="{{note.note}}"/>
+            <span class="input-group-addon">{{note.date | date:'yyyy-MM-dd'}}</span>
```
Comment (dting) [inline client/app/main/student/partial/interventions.html]: Let's change this date back to 'MM/dd/yy'

This matches 'shortDate' format for uib.

File: client/app/main/student/partial/interventions.html [hunk 11]
```diff
@@ -15,11 +15,9 @@
         <dt>Action Taken</dt>
         <dd>
           <div class="input-group">
-            <span class="input-group-btn">
-              <button type="button" class="btn btn-success" ng-click="open($index)">
-                <i class="fa fa-calendar"></i>
-              </button>
-            </span>
+            <input type="text" class="form-control truncate" readonly value="{{note.note}}"/>
+            <span class="input-group-addon">{{note.date | date:'yyyy-MM-dd'}}</span>
```
Comment (dting) [inline client/app/main/student/partial/interventions.html]: This doesn't look like it is merged correctly. You replace the action taken section with the notes section.
